lalrpop/src/build/mod.rs

```diff
@@ -1,562 +1,569 @@
-        grammar.prefix,
-        user_type_parameters,
-    );
-    rust!(
-        rust,
-        "fn to_triple(value: Self) -> Result<({L},{T},{L}), {parse_error}>;",
-        L = L,
-        T = T,
-        parse_error = parse_error,
-    );
-    rust!(rust, "}}");
-
-    rust!(rust, "");
-    if grammar.types.opt_terminal_loc_type().is_some() {
-        rust!(
-            rust,
-            "impl<{utp}> {p}ToTriple<{utp}> for ({L}, {T}, {L}) {{",
-            p = grammar.prefix,
-            utp = user_type_parameters,
-            L = L,
-            T = T,
-        );
-        rust!(
-            rust,
-            "fn to_triple(value: Self) -> Result<({L},{T},{L}), {parse_error}> {{",
-            L = L,
-            T = T,
-            parse_error = parse_error,
-        );
-        rust!(rust, "Ok(value)");
-        rust!(rust, "}}");
-        rust!(rust, "}}");
-
-        rust!(
-            rust,
-            "impl<{utp}> {p}ToTriple<{utp}> for Result<({L}, {T}, {L}), {E}> {{",
-            utp = user_type_parameters,
-            p = grammar.prefix,
-            L = L,
-            T = T,
-            E = E,
-        );
-        rust!(
-            rust,
-            "fn to_triple(value: Self) -> Result<({L},{T},{L}), {parse_error}> {{",
-            L = L,
-            T = T,
-            parse_error = parse_error,
-        );
-        rust!(rust, "match value {{");
-        rust!(rust, "Ok(v) => Ok(v),");
-        rust!(rust, "Err(error) => Err({p}lalrpop_util::ParseError::User {{ error }}),",
-              p = grammar.prefix);
-        rust!(rust, "}}"); // match
-        rust!(rust, "}}");
-        rust!(rust, "}}");
-    } else {
-        rust!(
-            rust,
-            "impl<{utp}> {p}ToTriple<{utp}> for {T} {{",
-            utp = user_type_parameters,
-            p = grammar.prefix,
-            T = T,
-        );
-        rust!(
-            rust,
-            "fn to_triple(value: Self) -> Result<((),{T},()), {parse_error}> {{",
-            T = T,
-            parse_error = parse_error,
-        );
-        rust!(rust, "Ok(((), value, ()))");
-        rust!(rust, "}}");
-        rust!(rust, "}}");
-
-        rust!(
-            rust,
-            "impl<{utp}> {p}ToTriple<{utp}> for Result<({T}),{E}> {{",
-            utp = user_type_parameters,
-            p = grammar.prefix,
-            T = T,
-            E = E,
-        );
-        rust!(
-            rust,
-            "fn to_triple(value: Self) -> Result<((),{T},()), {parse_error}> {{",
-            T = T,
-            parse_error = parse_error,
-        );
-        rust!(rust, "match value {{");
-        rust!(rust, "Ok(v) => Ok(((), v, ())),");
-        rust!(rust, "Err(error) => Err({p}lalrpop_util::ParseError::User {{ error }}),",
-              p = grammar.prefix);
-        rust!(rust, "}}"); // match
-        rust!(rust, "}}"); // fn
-        rust!(rust, "}}"); // impl
-    }
-
-    Ok(())
-}
+//! Utilies for running in a build script.
+
```

Comment:
what...is going on in this file?

Reply:
Apparently the previous version had CRLF terminators. I'll pull from this branch.